src/scripts/main.js

```diff
@@ -1,3 +1,36 @@
 'use strict';
 
 // write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const tableHeaders = document.querySelectorAll('th');
+  const tableBody = document.querySelector('tbody');
+
+  const tableRows = Array.from(tableBody.querySelectorAll('tr'));
+
+  tableHeaders.forEach((header, columnIndex) => {
+    header.addEventListener('click', () => {
+      const sortedRows = tableRows
+        .map((row) => {
+          const cellText = row.cells[columnIndex].textContent.trim();
+          const numericValue = parseFloat(cellText.replace(/[^0-9.-]+/g, ''));
+
+          return { row, cellText, numericValue };
+        })
+        .sort((a, b) => {
+          if (!isNaN(a.numericValue) && !isNaN(b.numericValue)) {
+            return a.numericValue - b.numericValue;
+          } else {
+            return a.cellText.localeCompare(b.cellText);
+          }
+        })
+        .map((item) => item.row);
```

Comment:
You're sorting the rows every time a header is clicked. This could be inefficient if the table is large. Consider a different approach where you keep track of the current sort order for each column, and simply reverse the order if the same column header is clicked again.